### PR TITLE
Extract script added

### PR DESCRIPTION
**Pull request by pphuangyi**
I added the `extract_toy-sp.py` to `scripts/`. 

I also added a `README_extract.md` documenting `toy1` data downloading and extraction.

Comments are welcomed!

### CODE REVIEW

**Comment by usert5432** [comment]
Hi Yi. This PR seem fine to me. The only thing I would like ask you, is to remove `toytools.egg-info`, `__pycache__`, ` .ipynb_checkpoints` directories from git please.

**Comment by brettviren** [comment]
> Hi Yi. This PR seem fine to me. The only thing I would like ask you, is to remove `toytools.egg-info`, `__pycache__`, ` .ipynb_checkpoints` directories from git please.

Yes, please make and commit a `.gitignore` file in every repo that blocks these kinds of unwanted files before they may be accidentally committed.

Here is the one I provided in `ls4gan-python` skeleton, mostly as-generated by "cookiecutter". 

https://github.com/LS4GAN/ls4gan-python/blob/master/.gitignore

Of course, adjust to taste and always one may do `git add -f <ignoredfile>` to override for specific files that would otherwise be ignored.

**Comment by pphuangyi** [comment]
Hi Brett and Dmitrii, I've added the `.gitignore` file so unwanted folders would show up again. Thank you for your suggestions!

File: scripts/extract_toy-sp.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	fnames = Path(f'{save_folder}/{member_folder}').glob(f'*.{file_extension}')
+	for fname in fnames:
+		shutil.move(fname, save_folder)
+	
+	# Remove the chain of parent folders (now empty)
+	path_base = os.path.normpath(member_folder).split(os.sep)[0]
+	shutil.rmtree(f'{save_folder}/{path_base}')
+
+def get_seed(fname):
+	return fname.split('-')[-1].split('_')[0]
```
Comment (YHRen) [inline scripts/extract_toy-sp.py]: add an example filename

File: scripts/extract_toy-sp.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		type	= str,
+	)
+
+	parser.add_argument(
+		'--seed_path',
+		'-s',
+		help	= 'The location for the seed folders. Images generated with seed=<seed> are contained in <seed_path>/seed-<seed>.',
+		dest	= 'seed_path',
+		type	= str,
+		default = None
```
Comment (YHRen) [inline scripts/extract_toy-sp.py]: equal signs are not aligned. @usert5432  will not like it :D

**Comment by usert5432** [comment]
Thank you Yi! There is one more thing about the git history. Right now, you have deleted binary files from the latest git revision, but these files are still saved in the git history. If we do nothing, then they will forever stay there, needlessly increasing the size of the git repository. So, it would be also good to overwrite the git history and remove binary files from it.

There are many ways to achieve that, but I guess the simplest will be to recreate this branch from scratch and add only files with code there?

**Review by YHRen — APPROVED** [review]
Good job! Thanks!

**Pull request merged**